### PR TITLE
feat(loadable component) : Support SSR using @loadable/component

### PR DESCRIPTION
**Pull request by Athira001**
# Description

This PR adds a helper method for SSR of components using @loadable/component
In order to render the chunk we used loadable component which basically used a loadable stats file created by webpack & from which we get to know which chunks need which JS.
As a starter, this is done as a part of the header component in Malibu

Fixes: https://github.com/quintype/ace-planning/issues/185

Documentation: will update post the production release

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: server/render.js [hunk 1]
```diff
@@ -31,3 +31,39 @@ exports.renderReduxComponent = function renderReduxComponent(
...
+  extractor,
+  props
+) {
+  const children = React.createElement(Component, props);
+  //Wrap your component using "collectChunks"
+  const component = extractor.collectChunks(
+    React.createElement(Provider, { store }, children)
+  );
+  // Render your Component
+  const renderString = ReactDOMServer.renderToString(component);
```
Comment (Jeevan-Kishore) [inline server/render.js:67]:
```suggestion
    return ReactDOMServer.renderToString(component);
```

**Review by Jeevan-Kishore — APPROVED** [review]

File: server/render.js [hunk 2]
```diff
@@ -31,3 +31,39 @@ exports.renderReduxComponent = function renderReduxComponent(
...
+ * @category Server
+ * @module render
+ */
+/**
+ * @param {Component} Component The Component to render
+ * @param {Redux} store The store to render
+ * @param {extractor} props Extractor from the statsFile
+ * @param {Object} props The props to pass to the component
+ */
+exports.renderLoadableReduxComponent = function renderLoadableReduxComponent(
```
Comment (sharangj) [inline server/render.js:56]: Please test this component.

File: server/render.js [hunk 3]
```diff
@@ -31,3 +31,39 @@ exports.renderReduxComponent = function renderReduxComponent(
...
+ * renderLoadableReduxComponent(Header, params.store, extractor)
+ * The changes to be done in the app level for rendering critical Css will be updated shortly
+ * ```
+ * @category Server
+ * @module render
+ */
+/**
+ * @param {Component} Component The Component to render
+ * @param {Redux} store The store to render
+ * @param {extractor} props Extractor from the statsFile
```
Comment (sharangj) [inline server/render.js:53]: Change to `Instance of ChunkExtractor from loadable.

File: server/render.js [hunk 4]
```diff
@@ -31,3 +31,39 @@ exports.renderReduxComponent = function renderReduxComponent(
...
+ */
+exports.renderLoadableReduxComponent = function renderLoadableReduxComponent(
+  Component,
+  store,
+  extractor,
+  props
+) {
+  const children = React.createElement(Component, props);
+  //Wrap your component using "collectChunks"
+  const component = extractor.collectChunks(
```
Comment (sharangj) [inline server/render.js:64]: What happens when there are multiple components. Would it add stats and strings to the same chunk?

File: server/render.js [hunk 5]
```diff
@@ -31,3 +31,39 @@ exports.renderReduxComponent = function renderReduxComponent(
...
+
+/**
+ * This namespace a helper method for SSR using @loadable/component
+ * ```javascript
+ * import { ChunkExtractor } from "@loadable/server";
+ * import {renderReduxComponent} from "@quintype/framework/server/render";
+ * import path from "path";
+ *
+ * const statsFile = path.resolve("stats.json"); // This is the stats file generated by webpack loadable plugin
+ * const extractor = new ChunkExtractor({ statsFile, entrypoints: ["topbarCriticalCss", "navbarCriticalCss"] }); // We create an extractor from the statsFile
```
Comment (sharangj) [inline server/render.js:43]: I think the chunk extractor needs to go inside the function.

Comment (nandap4790) [reply server/render.js:43]: It should be outside for the benefit of critical css

File: server/render.js (hunk 2 above)
Comment (nandap4790) [reply server/render.js:56]: done

File: server/render.js (hunk 4 above)
Comment (nandap4790) [reply server/render.js:64]: Multiple chunks are created

File: server/render.js (hunk 3 above)
Comment (nandap4790) [reply server/render.js]: Done

File: test/integration/render-test.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import { ChunkExtractor } from "@loadable/server";
+import path from "path";
+import { renderLoadableReduxComponent } from "../../server/render";
+import { createQtStore } from "../../store/create-store";
+import { TestComponent } from "../data/test-component";
+const assert = require("assert");
+
+describe("RenderLoadableReduxComponent", () => {
+  it("Renders the component by passing the stats and entry points ", () => {
```
Comment (sharangj) [inline test/integration/render-test.js:9]: Please write a test for multiple components also.

Comment (sharangj) [reply test/integration/render-test.js:9]: What happens if we use multiple instances of renderLoadableReduxComponent.

**Pull request merged**